### PR TITLE
Added Hazelcast Jet helm chart

### PR DESCRIPTION
**Pull request by eminn**

### CODE REVIEW

File: stable/hazelcast-jet/README.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            <discovery-strategies>
+              <discovery-strategy enabled="true" class="com.hazelcast.kubernetes.HazelcastKubernetesDiscoveryStrategy">
+              </discovery-strategy>
+            </discovery-strategies>
+          </join>
+        </network>
+        <!-- Custom Configuration Placeholder -->
+      </hazelcast>
+
+      hazelcast-jet.xml: |-
```
Comment (leszko) [inline stable/hazelcast-jet/README.md]: formatting: I think you have too many (white)spaces before `hazelcast-jet.xml`.

File: stable/hazelcast-jet/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    
+        <properties>
+          <property name="hazelcast.discovery.enabled">true</property>
+        </properties>
+        <network>
+          <join>
+            <multicast enabled="false"/>
+            <tcp-ip enabled="false" />
+            <discovery-strategies>
+              <discovery-strategy enabled="true" class="com.hazelcast.kubernetes.HazelcastKubernetesDiscoveryStrategy">
```
Comment (leszko) [inline stable/hazelcast-jet/README.md]: Don't you want / need to add the `service-name` and `namespace` properties, like we did in the `hazelcast/hazelcast` Helm Chart?

File: stable/hazelcast-jet/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+        ports:
+        - name: hazelcast-jet
+          containerPort: 5701
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
+          httpGet:
+            path: /hazelcast/health/node-state
```
Comment (leszko) [inline stable/hazelcast-jet/templates/statefulset.yaml]: Don't you have some other health check for Jet? Meaning: is it possible that Hazelcast is up, but Jet is down?

File: stable/hazelcast-jet/values.yaml [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+image:
+  # repository is the Hazelcast Jet image name
+  repository: "hazelcast/hazelcast-jet"
+  # tag is the Hazelcast Jet image tag
+  tag: "0.7"
+  # pullPolicy is the Docker image pull policy
+  # It's recommended to change this to 'Always' if the image tag is 'latest'
+  # ref: http://kubernetes.io/docs/user-guide/images/#updating-images
+  #
+  pullPolicy: Always
```
Comment (leszko) [inline stable/hazelcast-jet/values.yaml]: How about chaning to `IfNotPresent`? I think `Always` is not recommended by default.

Comment (mesutcelik) [reply stable/hazelcast-jet/values.yaml]: I agree with @leszko . Always is only recommended if image tag is `latest` by default which is not the case for our helm charts.

File: stable/hazelcast-jet/templates/NOTES.txt [hunk 5]
```diff
@@ -0,0 +1,58 @@
+** Hazelcast Jet cluster is being deployed! **
```
Comment (mesutcelik) [inline stable/hazelcast-jet/templates/NOTES.txt]: is there a REST call to do some smoke testing after the deployment. That looks like a common practice in helm deployment stdout messages.

**Comment by mesutcelik** [comment]
@eminn @leszko ,
Do you think we need to create an abstract helm chart and introduce as dependency to our helm charts? Our number of helm charts are increasing so maintenance can be an issue over time.

What do you think?

**Comment by leszko** [comment]
@mesutcelik what do you mean about "abstract"? I think if you make hazelcast-jet dependent on hazelcast, then it's kind-of different to what Emin sent right now:
- Dependent Helm Chart - Helm installs Hazelcast Cluster (and then Jet can use it as a Client?)
- The way it is right now - Hazelcast is kind-of embedded into Hazelcast Jet

I think we want the former one.

Saying that, Mesut, I agree with you. We have too many Helm Chart and the maintenance becomes difficult. I think we should think about some automation over this. Like that you change once and all the PRs to all repos generates. Or something like that.

**Comment by eminn** [comment]
@mesutcelik @leszko I see the pain point of having multiple helm charts.

We might need to create a PoC Jet chart which uses IMDG helm chart as a dependency to see whether this is possible or not.

File: stable/hazelcast-jet/templates/NOTES.txt (hunk 5 above)
Comment (eminn) [reply stable/hazelcast-jet/templates/NOTES.txt]: There is no Jet specific REST endpoints. Maybe we can use IMDG health check REST endpoint. WDYT?

**Comment by mesutcelik** [comment]
When I say `abstract`. I meant some kind of core helm chart which can or can't be used alone and all the other charts has to depend on it. It will mostly contain deployment configs where we copy paste between helm charts...

I have no clear idea how to do it but I see the requirement over time.

**Comment by leszko** [comment]
hmm, I don't think that there is something like `abstract` in Helm Charts. There is an `include` keyword and we could write some `*.tpl` stuff, but I guess it will just complicate things. For us and for the users.

I'd rather think how to keep the Helm Charts synchronized. Like maybe a script which takes a relative git patch and creates PRs everywhere? I'll think about it.

Any idea here would be fine. Nobody likes the maintenance :)

**Comment by mesutcelik** [comment]
👍

**Pull request merged**